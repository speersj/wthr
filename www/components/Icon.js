import React from 'react'
import PropTypes from 'prop-types'

Icon.propTypes = { name: PropTypes.string.isRequired }

export default function Icon({ name }) {
  return (
    <i
      title={name}
      role="img"
      data-testid="plain-icon"
      className={iconClassName(name)}
    />
  )
}

function iconClassName(iconName) {
  return `wi wi-${iconName}`
}
