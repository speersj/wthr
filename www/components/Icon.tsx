import React from "react";

export default function Icon({ name }: { name: string }) {
  return (
    <i
      title={name}
      role="img"
      data-testid="plain-icon"
      className={iconClassName(name)}
    />
  );
}

function iconClassName(iconName: string) {
  return `wi wi-${iconName}`;
}
