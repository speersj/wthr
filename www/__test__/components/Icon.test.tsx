import React from "react";
import { render } from "react-testing-library";
import Icon from "../../components/Icon";

describe("Icon", () => {
  it("renders the title attribute", () => {
    render(<Icon name="test" />).getByTitle("test");
  });

  it("renders an <i> tag", () => {
    const el = render(<Icon name="test" />).getByTitle("test");
    expect(el.tagName).toEqual("I");
  });

  it("renders with the correct role (img)", () => {
    render(<Icon name="test" />).getByRole("img");
  });

  it("renders the correct classnames", () => {
    const el = render(<Icon name="test" />).getByTitle("test");
    expect(el.className).toEqual("wi wi-test");
  });

  it("outputs correct html", () => {
    expect(render(<Icon name="test" />).getByTitle("test")).toMatchSnapshot();
  });
});
