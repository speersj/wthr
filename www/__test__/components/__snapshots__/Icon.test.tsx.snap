// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon outputs correct html 1`] = `
<i
  class="wi wi-test"
  data-testid="plain-icon"
  role="img"
  title="test"
/>
`;
